ios/brave-ios/Sources/Brave/Frontend/Browser/TabObserver.swift

```diff
@@ -39,15 +56,21 @@ extension TabObserver {
   func tabDidChangeVisibleSecurityState(_ tab: Tab) {}
   func tabDidChangeBackForwardState(_ tab: Tab) {}
   func tabDidChangeSampledPageTopColor(_ tab: Tab) {}
+  func tabWillBeDestroyed(_ tab: Tab) {}
```

Comment:
can we not have a default implementation for `tabWillBeDestroyed` as a reminder to not forget to implement necessary clean-house actions like removing observers?

Reply:
We could technically, though its not absolutely necessary unless you intend to observe for the entire Tab's lifetime?

Reply:
yeah just a suggestion. im not too sure how often we need to observe the tab's lifetime.

Reply:
I'll leave as is for now and if we find ourselves forgetting too often we can remove the default. There is technically no UB if you forget to cleanup observers/policy deciders since the weak points will nil out, other than a few wasted cpu cycles running on those empty AnyTabObservers